### PR TITLE
change image registry from dockerhub to ibm container registry

### PR DESCRIPTION
**Pull request by Basavaraju-G**
# Changes

Move test images from free dockerhub account to ibm container registry.

# Submitter Checklist

As the author of this PR, please check off the items in this checklist:

- [ ] Has [Docs](https://github.com/tektoncd/community/blob/main/standards.md#docs) included if any changes are user facing
- [ ] Has [Tests](https://github.com/tektoncd/community/blob/main/standards.md#tests) included if any functionality added or changed
- [x] Follows the [commit message standard](https://github.com/tektoncd/community/blob/main/standards.md#commits)
- [x] Meets the [Tekton contributor standards](https://github.com/tektoncd/community/blob/main/standards.md) (including
  functionality, content, code)
- [x] Has a kind label. You can add one by adding a comment on this PR that contains `/kind <type>`. Valid types are bug, cleanup, design, documentation, feature, flake, misc, question, tep
- [ ] Release notes block below has been updated with any user facing changes (API changes, bug fixes, changes requiring upgrade notices or deprecation warnings)
- [ ] Release notes contains the string "action required" if the change requires additional action from users switching to the new release

# Release Notes

```release-note
NONE
```

### CODE REVIEW

**Comment by Basavaraju-G** [comment]
Earlier multi-arch test images available in ibmcom workspace (i.e https://hub.docker.com/r/ibmcom/) which is sponsored by cloud pak team. due to security reasons, IBM CISO has issued new guidelines to not publish images on dockerhub/ibmcom namespace. alternative one to use ibm public cloud container registry to host images.

**Comment by lbernick** [comment]
Thanks @Basavaraju-G! Would you mind linking this guidance from IBM CISO if it's publicly available? Could you also create an issue for this? If other tekton projects use dockerhub/ibmcom we should migrate them as well and can use the issue for that.

Re: which alternative to dockerhub-- I think this should probably be reviewed by someone with more knowledge of our infrastructure (maybe @afrittoli?) but I have a few questions:
- would it make sense to use GCR, since we're already publishing images there?
- do we need to maintain any additional credentials for ICR?
- do we know of any security reasons to prefer any cloud provider's container registry?

**Comment by Basavaraju-G** [comment]
> * would it make sense to use GCR, since we're already publishing images there?
yes possible, but we need to have access GCR registry to build for s390x/ppc64le platform image.
> * do we need to maintain any additional credentials for ICR?
Not required, No authentication is required to pull the images.
> * do we know of any security reasons to prefer any cloud provider's container registry?
i will check this with @afrittoli on this.

File: test/multiarch_utils.go [hunk 1]
```diff
@@ -56,22 +56,20 @@ func initImageNames() map[int]string {
...
-			kanikoImage:    "ibmcom/kaniko-project-executor-ppc64le:v0.17.1",
-			dockerizeImage: "ibmcom/dockerize-ppc64le",
+			dockerizeImage: "icr.io/tekton-upstream/dockerize-ppc64le",
 		}
 	default:
 		return map[int]string{
 			busyboxImage:   "busybox@sha256:895ab622e92e18d6b461d671081757af7dbaa3b00e3e28e12505af7817f73649",
 			registryImage:  "registry",
-			kanikoImage:    "gcr.io/kaniko-project/executor:v1.3.0",
+			kanikoImage:    "gcr.io/kaniko-project/executor:v1.9.2",
```
Comment (afrittoli) [inline test/multiarch_utils.go:72]: I wonder why the kaniko image was removed from the s390x and ppc64le ?

File: test/examples_test.go [hunk 2]
```diff
@@ -291,10 +291,10 @@ func imageNamesMapping() map[string]string {
 			"registry":                              getTestImage(registryImage),
 			"node":                                  "node:alpine3.11",
 			"gcr.io/cloud-builders/git":             "alpine/git:latest",
-			"docker:dind":                           "ibmcom/docker-s390x:20.10",
+			"docker:dind":                           "icr.io/tekton-upstream/docker-s390x:20.10",
```
Comment (afrittoli) [inline test/examples_test.go:294]: NIT: these docker images are not owned and maintained by the Tekton community, so the `tekton-upstream` project name is slightly misleading.  I think it's fine to keep it, but perhaps a comment about the source / maintainer of the images would be nice

**Review by afrittoli — COMMENTED** [review]
Thank you for this!

File: test/examples_test.go (hunk 2 above)
Comment (Basavaraju-G) [reply test/examples_test.go:294]: Thanks, i will add maintainer details

**Comment by Basavaraju-G** [comment]
as of now test images in ibmcom are going to work as it is. so new images we will be pushing to new icr registry. so closing this PR.

**Pull request closed without merging**